python/tk_multi_publish2/api/item.py

```diff
@@ -406,6 +402,10 @@ def get_thumbnail_as_path(self):
         if self.thumbnail is None:
             return None
 
+        # nothing to do if running without a UI
+        if not sgtk.platform.current_engine().has_ui:
+            return None
```

Comment:
Why would we even "disable" this function if no UI?
I don't see anything in the code below that requires QT. Would `thumbnail.save(temp_path)` require a UI?

Could you dig a bit in the code history to see why this test was introduced?

Reply:
Basically, the thumbnail attribute is a QtGui.QPixmap => https://github.com/shotgunsoftware/tk-multi-publish2/blob/SHOT-4192-thumbnail-in-bg-mode/python/tk_multi_publish2/api/item.py#L892-L925

I've refactored my code a little bit to make sure we're checking the mode before calling the thumbnail attribute even if there is a check in the getter property to make sure we can manipulate Qt pixmap object